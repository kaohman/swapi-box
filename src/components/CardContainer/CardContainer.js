import React from 'react';
import Card from '../Card/Card';
import PropTypes from 'prop-types';

const CardContainer = ({cards, favorites, toggleFavorite}) => {
  return (
    <div id='card-container'>
      <Card 
        data={cards} 
        toggleFavorite={toggleFavorite} 
        favorites={favorites}
      />
    </div>
  );
}

CardContainer.propTypes = {
  cards: PropTypes.array
}

CardContainer.defaultProps = {
  cards: []
}

export default CardContainer;