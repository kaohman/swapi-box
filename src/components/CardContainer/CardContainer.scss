#card-container {
  margin: 20px auto;
  padding: 15px;
  width: 80%;
  height: 460px;
  overflow: scroll;
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(250px, 1fr));
  justify-content: center;
  align-content: flex-start;
  grid-gap: 20px;
}

#show-button-div {
  width: 250px;
  height: 300px;
  margin: auto;
}

.show-button {
  display: block;
  margin: 0 auto;
}