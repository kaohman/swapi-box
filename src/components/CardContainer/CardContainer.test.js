import React from 'react';
import ReactDOM from 'react-dom';
import CardContainer from './CardContainer';
import { shallow } from 'enzyme';

describe('CardContainer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <CardContainer
        cards={cardsToDisplay}
        favorites={favorites}
        toggleFavorite={this.toggleFavorite}
        currentPage={currentPage}
      />
    );
  });

  it('should match the snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });
});