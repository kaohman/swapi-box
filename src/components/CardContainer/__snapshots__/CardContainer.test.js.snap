// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot with all data passed in 1`] = `
<div
  id="card-container"
>
  <Card
    data={Array []}
  />
</div>
`;
